x/bank/test_helpers.go

```diff
@@ -0,0 +1,129 @@
+package bank
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/mock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/crypto"
+)
+
+// ModuleInvariants runs all invariants of the bank module.
+// Currently only runs non-negative balance invariant
+func ModuleInvariants(t *testing.T, app *mock.App, log string) {
```

Comment:
Would it make sense for debugging to somehow have a test object? The reason I ask this is because require/asserts provide useful traces / line numbers, whereas if we just passed errors back we'd have to pinpoint the line via error message.

Reply:
Good question. I would ask @xla for some thoughts

Reply:
Two observations, this file indicates it is meant for use in the context of testing, yet the file is always compiled as it doesn't have the `_test` suffix. And I agree it doesn't sound like a helper it shuld be a broken out with a proper signature and then this func reworked into a helper which takes the `testing.T`. But it could be that my comment is irrelevant as this part as been changed so diregrad, just some drive by commenting. :pewpew: